packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,672 @@
+
+#### Designs
+
+- [Figma - Select](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=13590%3A171090&t=KTWGvEUBEuUdQh9K-0)
+
+#### API
+
+Sample Usage
+
+```jsx
+<Dropdown.SelectInput
+  icon={CalendarIcon}
+  label="Your favorite Design System"
+  helpText="Select only one"
+/>
+```
+
+Props
+
+Mostly similar to other `Input` components extended from `BaseInput`
+
+```ts
+type SelectInputProps = {
+  // ... FormLabelProps like `label`, `labelPosition`
+  // ... FormValidationProps like `validationState`, `helpText`, `errorText`, etc.
+  // ... State Props like `isRequired`, `isDisabled`,
+  icon: IconComponent;
+};
+```
+
+#### A11y
+
+Would be ideal if we can render it as `button` with `aria-expanded` prop to handle state of dropdown open and close
+
+### `Dropdown.Overlay`
+
+An Overlay that positions dropdown correctly.
+
+The animations for dropdown and Aria attributes like `aria-multiselectable="true"` `role="listbox"` like things can be added on Dropdown overlay here based on variant passed to `Dropdown` component.
+
+#### API
+
+```jsx
+<Dropdown>
+  <Dropdown.SelectInput label="Click For Greetings" />
+  <Dropdown.Overlay>Hi</Dropdown.Overlay>
+</Dropdown>
+```
+
+It won't accept any props apart from children.
+
+<br/><br/>
+
+## ActionList
+
+Plain list that can be rendered inside dropdown, in navigation menu, or anywhere. It has selectable items that perform some action on click. It can also have multiple selectable items that render as checkboxes
+
+<img src="2022-11-23-09-50-05.png" alt="ActionList figma screenshot" width="300" />
+
+### Design
+
+- [Figma - ActionList]()
+
+### API
+
+#### Simplest Usage Possible
+
+```jsx
+<ActionList
+  backgroundLevel={2}
+  onChange={(e) => {
+    console.log({ name: e.name, values: e.values });
+  }}
+>
+  <ActionList.Item
+    title="Accept API"
+    value="accept"
+    desciption="Accept this API"
+    leading={<StarIcon />}
+    trailing={<Text>⌘ + Space + A</Text>}
+    onClick={() => {}}
+  />
+  <ActionList.Item
+    title="Delete"
+    desciption="Delete this value"
+    value="delete"
+    intent="negative"
+    leading={<TrashIcon />}
+    trailing={<Text>⌘ + Space + D</Text>}
+    onClick={() => {}}
+  />
+</ActionList>
+```
+
+#### Most Complex Usage Possible
+
+```jsx
+<ActionList backgroundLevel={2}>
+  <ActionList.Header title="Recent Searches" icon={HistoryIcon} />
+  <ActionList.Section title="Section Heading">
+    <ActionList.Item
+      title="Accept API"
+      value="accept"
+      desciption="Accept this API"
+      leading={<StarIcon />}
+      trailing={<Text>⌘ + Space</Text>}
+    />
+  </ActionList.Section>
+  <ActionList.Item
+    title="Delete"
+    desciption="Delete this value"
+    value="delete"
+    intent="negative"
+    leading={<TrashIcon />}
+    trailing={<Text>⌘ + Space + D</Text>}
+  />
+  <ActionList.Footer
+    title="Footer Title"
+    description="Footer Description"
+    leading={<DocIcon />}
```

Comment:
The reason I am calling it just `leading` is, `leadingVisual` sort of conveys that it's a "visual" which I would include icons, assets, images, etc. But in trailing for example we have a keyboard binding text which is a text which might not be just a visual.

Also in footer, we have `trailing` prop where we can pass an entire JSX of action buttons so that is also something I wouldn't consider as just "visual". Calling it `leading` and `trailing` leaves more space for us to generically use it.